discord/appinfo.py

```diff
@@ -360,3 +544,86 @@ class AppInstallParams:
     def __init__(self, data: AppInstallParamsPayload) -> None:
         self.scopes: list[str] = data.get("scopes", [])
         self.permissions: Permissions = Permissions(int(data["permissions"]))
+
+    def _to_payload(self) -> dict[str, object]:
+        """Serialize this object into an application install params payload.
+
+        Returns
+        -------
+        Dict[str, Any]
+            A dict with ``scopes`` and ``permissions`` (string form) suitable for the API.
+        """
+        if self.permissions.value > 0 and "bot" not in self.scopes:
+            raise ValueError(
+                "'bot' must be in install_params.scopes if permissions are requested"
+            )
+        return {
+            "scopes": list(self.scopes),
+            "permissions": str(self.permissions.value),
+        }
+
+
+class IntegrationTypesConfig:
+    """Represents per-installation context configuration for an application.
+
+    This object is used to build the payload for the ``integration_types_config`` field when editing an application.
+
+    Parameters
+    ----------
+    guild: Optional[:class:`AppInstallParams`]
+        The configuration for the guild installation context. Omit to leave unchanged; pass ``None`` to clear.
+    user: Optional[:class:`AppInstallParams`]
+        The configuration for the user installation context. Omit to leave unchanged; pass ``None`` to clear.
+    """
+
+    __slots__ = ("guild", "user")
+
+    def __init__(
+        self,
+        *,
+        guild: AppInstallParams | None = utils.MISSING,
+        user: AppInstallParams | None = utils.MISSING,
+    ) -> None:
+        self.guild = guild
+        self.user = user
+
+    @staticmethod
+    def _get_ctx(raw: dict | None, key: int):
+        if raw is None:
+            return None
+        if key in raw:
+            return raw[key]
+        skey = str(key)
+        return raw.get(skey)
+
+    @staticmethod
+    def _decode_ctx(value: dict | None) -> AppInstallParams | None:
```

Comment:
```suggestion
    def _decode_ctx(value: dict[Any, Any] | None) -> AppInstallParams | None:
```
Or other relevant typing, I don't have the context to know here